solidity/test/bridge/Bridge.Frauds.test.ts

```diff
@@ -1305,11 +1345,334 @@ describe("Bridge - Fraud", () => {
+                  additionalAssertions: async (args: any) => {
+                    it("should decrease the live wallets count", async () => {
+                      expect(await bridge.liveWalletsCount()).to.be.equal(0)
+                    })
+
+                    it("should not unset the active wallet", async () => {
+                      expect(
+                        await bridge.activeWalletPubKeyHash()
+                      ).to.be.not.equal(
+                        "0x0000000000000000000000000000000000000000"
+                      )
+                    })
+                  },
+                },
+                {
+                  testName:
+                    "when wallet state is Live and the wallet is the active one",
+                  walletState: walletState.Live,
+                  additionalSetup: async () => {
+                    await bridge.setActiveWallet(walletPublicKeyHash)
+                  },
+                  additionalAssertions: async (args: any) => {
+                    it("should decrease the live wallets count", async () => {
+                      expect(await bridge.liveWalletsCount()).to.be.equal(0)
+                    })
+
+                    it("should unset the active wallet", async () => {
+                      expect(await bridge.activeWalletPubKeyHash()).to.be.equal(
+                        "0x0000000000000000000000000000000000000000"
+                      )
+                    })
+                  },
+                },
+                {
+                  testName: "when wallet state is MovingFunds",
+                  walletState: walletState.MovingFunds,
+                  additionalSetup: async () => {},
+                  additionalAssertions: async () => {},
+                },
+                {
+                  testName: "when wallet state is Closing",
+                  walletState: walletState.Closing,
+                  additionalSetup: async () => {},
+                  additionalAssertions: async () => {},
+                },
+              ]
+
+              testData.forEach((test) => {
+                context(test.testName, async () => {
+                  let tx: ContractTransaction
+
+                  before(async () => {
+                    await createSnapshot()
+
+                    await bridge.setWallet(walletPublicKeyHash, {
+                      ...walletDraft,
+                      state: test.walletState,
+                    })
+
+                    await bridge
+                      .connect(thirdParty)
+                      .submitFraudChallenge(
+                        walletPublicKey,
+                        data.sighash,
+                        data.signature,
+                        {
+                          value: fraudChallengeDepositAmount,
+                        }
+                      )
+
+                    await increaseTime(fraudChallengeDefeatTimeout)
+
+                    await test.additionalSetup()
```

Comment:
Shouldn't we execute it before `submitFraudChallenge` at line 1440? The additional setup we execute is about setting an active wallet, and it happens at the very beginning of wallet creation.